AferoSwiftSDK/Core/Models/OfflineSchedule.swift

```diff
@@ -883,13 +978,21 @@ open class OfflineSchedule: NSObject {
                 }
             }
 
-            public init(dayOfWeek: DayOfWeek = .sunday, hour: Hour = 0, minute: Minute = 0, flags: Flags = [.usesDeviceTimeZone]) {
-                self.dayOfWeek = dayOfWeek
+            
+            public init(daysOfWeek: Set<DayOfWeek> = [], hour: Hour = 0, minute: Minute = 0, flags: Flags = [.usesDeviceTimeZone], useCompactDayRepresentation: Bool = false ) {
+                self._daysOfWeek = daysOfWeek
+                if (useCompactDayRepresentation || daysOfWeek.count > 1) {
```

Comment:
Can be compressed since we don't need to look at `useCompactDayRepresentation` if `daysOfWeek.count > 1`.

```swift
self.useCompactDayRepresentation = daysOfWeek.count > 1 || useCompactDayRepresentation;
```